### PR TITLE
feat: add azure linux ingest parser

### PR DESCRIPTION
**Pull request by LeoArurault**
Migrate Azure Linux intake to Ingest. I reused same test files.

### CODE REVIEW

File: Azure/azure-linux/ingest/parser.yml [hunk 1]
```diff
@@ -0,0 +1,282 @@
+name: untitled-name
+pipeline:
+  - name: json_event
+    description: ''
+    filter: ''
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:5]:
```suggestion
```

Are these fields mandatory?

File: Azure/azure-linux/ingest/parser.yml [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+    external:
+      name: json.parse-json
+      properties:
+        input_field: '{{original.message}}'
+        output_field: message
+  - name: stage2
+    description: ''
+    filter: ''
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:13]:
```suggestion
```

File: Azure/azure-linux/ingest/parser.yml [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+    external: null
+  - name: grok_sshd
+    description: ''
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:16]:
```suggestion
```

File: Azure/azure-linux/ingest/parser.yml [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+      {{json_event.message.properties.ident == "systemd" or
+      json_event.message.properties.ident == "sudo"}}
+    external: null
+stages:
+  stage2:
+    actions:
+      - set:
+          log.level: '{{json_event.message.level}}'
+          process.pid: '{{json_event.message.properties.pid}}'
+          process.id: '{{json_event.message.properties.pid}}'
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:172]:
```suggestion
```

`process.id` is not a valid ECS field.

File: Azure/azure-linux/ingest/parser.yml [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+          log.hostname: '{{json_event.message.properties.hostname}}'
+          event.created: '{{json_event.message.time}}'
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:174]: This field is generally handled by the log collector (here, `oc-intake-axens`). This is the time at which the event was received. You can probably update the `@timestamp` field to change the date of the event itself.

File: Azure/azure-linux/ingest/parser.yml [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+          event.message: '{{json_event.message.properties.Msg}}'
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:175]: `event.message` is not a valid ECS field.

File: Azure/azure-linux/ingest/parser.yml [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+    actions:
+      - set:
+          user.name: '{{ grok_crond.result.user_name }}'
+          process.args: '{{ grok_crond.result.process_args.split() }}'
+          process.executable: '{{ grok_crond.result.process_executable }}'
+        filter: ''
+        name: set
+      - set:
+          process.executable: /bin/sh
+        filter: '{{ grok_crond.result.process_executable == none }}'
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:230]: Isn't it hazardous to infer the executable name here?

**Review by otetard — CHANGES_REQUESTED** [review]
We should be as near as possible from the official ECS specifications.

(Didn't reviewed the whole parser logic tough.)

File: Azure/azure-linux/ingest/parser.yml (hunk 1 above)
Comment (LeoArurault) [reply Azure/azure-linux/ingest/parser.yml:5]: The parser was creating using the custom intake UI, which create some fields even if they are empty. I will remove them

File: Azure/azure-linux/ingest/parser.yml (hunk 7 above)
Comment (LeoArurault) [reply Azure/azure-linux/ingest/parser.yml:230]: It was done in the original parser, but I can remove it

File: Azure/azure-linux/ingest/parser.yml (hunk 6 above)
Comment (LeoArurault) [reply Azure/azure-linux/ingest/parser.yml:156]: Moved to `azure_linux.event`

**Review by PierrickV — DISMISSED** [review]

File: Azure/azure-linux/ingest/parser.yml (hunk 7 above)
Comment (otetard) [reply Azure/azure-linux/ingest/parser.yml:201]: Considering that the process is `/bin/sh` is probably not true in all situations.

IMHO, you’d better put the whole command line to the `process.command_line` and not try to infer either `process.args` and `process.executable`.

File: Azure/azure-linux/ingest/parser.yml [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+      - set:
+          action.name: systemd:session
+          action.type: open
+        filter: '{{ grok_systemd.result.systemd_observation == "Started Session" }}'
+        name: set
+  crond:
+    actions:
+      - set:
+          user.name: '{{ grok_crond.result.user_name }}'
+          process.args: '{{ grok_crond.result.process_args.split() }}'
```
Comment (otetard) [inline Azure/azure-linux/ingest/parser.yml:196]: Not sure that is is a good idea to use `split()` here. Consider the following command line for example:

```
/bin/sh -c "while true; do sleep 10 ; done"
```

Here, arguments should be the following thing:

```json
[
  "/bin/sh",
  "-c",
  "while true; do sleep 10 ; done"
]
```

I think that inferring arguments is petty difficult to do. I think that we dropped support for trying to extract args in our Linux parsers.

File: Azure/azure-linux/ingest/parser.yml (hunk 7 above)
Comment (LeoArurault) [reply Azure/azure-linux/ingest/parser.yml]: Changed

File: Azure/azure-linux/ingest/parser.yml (hunk 8 above)
Comment (LeoArurault) [reply Azure/azure-linux/ingest/parser.yml]: split removed

File: Azure/azure-linux/tests/cron_command1.json [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+            "outcome": "success"
+        },
+        "azure_linux": {
+            "message": "(root) CMD (/usr/lib64/sa/sa1 1 1)"
+        },
+        "log": {"hostname": "LinuxRedhatDesktop", "level": "info"},
+        "message": "{ \"time\" : \"2019-06-27T14:50:01.0000000Z\",\"resourceId\" : \"/subscriptions/128ed5ce-4f50-4b5f-a3b0-08233b5a86b6/resourceGroups/demo.sekoia.io/providers/Microsoft.Compute/virtualMachines/LinuxRedhatDesktop\",\"properties\" : {\"ident\" : \"CROND\",\"pid\" : \"21188\",\"Ignore\" : \"syslog\",\"Facility\" : \"cron\",\"Severity\" : \"info\",\"EventTime\" : \"2019-06-27T14:50:01+0000\",\"SendingHost\" : \"localhost\",\"Msg\" : \"(root) CMD (/usr/lib64/sa/sa1 1 1)\",\"hostname\" : \"LinuxRedhatDesktop\",\"FluentdIngestTimestamp\" : \"2019-06-27T14:50:01Z\"},\"category\" : \"cron\",\"level\" : \"info\",\"operationName\" : \"LinuxSyslogEvent\"}",
+        "process": {
+            "executable": "/usr/lib64/sa/sa1",
+            "args": "1 1",
```
Comment (otetard) [inline Azure/azure-linux/tests/cron_command1.json:23]:
```suggestion
            "command_line": "/usr/lib64/sa/sa1 1 1",
```

File: Azure/azure-linux/tests/omsagent_command.json [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        },
+        "log": {"hostname": "LinuxRedhatDesktop", "level": "info"},
+        "message": "{ \"time\" : \"2019-06-27T14:48:18.0000000Z\",\"resourceId\" : \"/subscriptions/128ed5ce-4f50-4b5f-a3b0-08233b5a86b6/resourceGroups/demo.sekoia.io/providers/Microsoft.Compute/virtualMachines/LinuxRedhatDesktop\",\"properties\" : {\"ident\" : \"sudo\",\"Ignore\" : \"syslog\",\"Facility\" : \"authpriv\",\"Severity\" : \"notice\",\"EventTime\" : \"2019-06-27T14:48:18+0000\",\"SendingHost\" : \"localhost\",\"Msg\" : \"omsagent : TTY=unknown ; PWD=/opt/microsoft/omsconfig/Scripts/2.6x-2.7x ; USER=root ; COMMAND=/opt/microsoft/omsconfig/Scripts/OMSYumUpdates.sh\",\"hostname\" : \"LinuxRedhatDesktop\",\"FluentdIngestTimestamp\" : \"2019-06-27T14:48:18Z\"},\"category\" : \"authpriv\",\"level\" : \"notice\",\"operationName\" : \"LinuxSyslogEvent\"}",
+        "user": {
+            "name": "root"
+        },
+        "process": {
+            "executable": "/opt/microsoft/omsconfig/Scripts/OMSYumUpdates.sh",
+            "working_directory": "/opt/microsoft/omsconfig/Scripts/2.6x-2.7x"
```
Comment (otetard) [inline Azure/azure-linux/tests/omsagent_command.json:27]:
```suggestion
            "working_directory": "/opt/microsoft/omsconfig/Scripts/2.6x-2.7x",
            "command_line": "/opt/microsoft/omsconfig/Scripts/OMSYumUpdates.sh"
```

File: Azure/azure-linux/tests/omsagent_command2.json [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+        },
+        "message": "{ \"time\" : \"2019-07-02T13:46:15.0000000Z\",\"resourceId\" : \"/subscriptions/128ed5ce-4f50-4b5f-a3b0-08233b5a86b6/resourceGroups/demo.sekoia.io/providers/Microsoft.Compute/virtualMachines/LinuxRedhatDesktop\",\"properties\" : {\"ident\" : \"sudo\",\"Ignore\" : \"syslog\",\"Facility\" : \"authpriv\",\"Severity\" : \"notice\",\"EventTime\" : \"2019-07-02T13:46:15+0000\",\"SendingHost\" : \"localhost\",\"Msg\" : \"omsagent : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/test -r /var/lib/docker/containers/bf64bddcdb7d18a3090980d2539e2c15c924138f489c280871941064850f7d16/bf64bddcdb7d18a3090980d2539e2c15c924138f489c280871941064850f7d16-json.log\",\"hostname\" : \"LinuxRedhatDesktop\",\"FluentdIngestTimestamp\" : \"2019-07-02T13:46:15Z\"},\"category\" : \"authpriv\",\"level\" : \"notice\",\"operationName\" : \"LinuxSyslogEvent\"}",
+        "user": {
+            "name": "root"
+        },
+        "process": {
+            "executable": "/bin/test",
+            "args": "-r /var/lib/docker/containers/bf64bddcdb7d18a3090980d2539e2c15c924138f489c280871941064850f7d16/bf64bddcdb7d18a3090980d2539e2c15c924138f489c280871941064850f7d16-json.log",
```
Comment (otetard) [inline Azure/azure-linux/tests/omsagent_command2.json:29]:
```suggestion
            "command_line": "/bin/test -r /var/lib/docker/containers/bf64bddcdb7d18a3090980d2539e2c15c924138f489c280871941064850f7d16/bf64bddcdb7d18a3090980d2539e2c15c924138f489c280871941064850f7d16-json.log",
```

File: Azure/azure-linux/tests/cron_command2.json [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        },
+        "message": "{ \"time\" : \"2019-06-27T14:29:01.0000000Z\",\"resourceId\" : \"/subscriptions/128ed5ce-4f50-4b5f-a3b0-08233b5a86b6/resourceGroups/demo.sekoia.io/providers/Microsoft.Compute/virtualMachines/LinuxRedhatDesktop\",\"properties\" : {\"ident\" : \"CROND\",\"pid\" : \"16373\",\"Ignore\" : \"syslog\",\"Facility\" : \"cron\",\"Severity\" : \"info\",\"EventTime\" : \"2019-06-27T14:29:01+0000\",\"SendingHost\" : \"localhost\",\"Msg\" : \"(root) CMD ([ -f /etc/krb5.keytab ] && [ \\\\( ! -f /etc/opt/omi/creds/omi.keytab \\\\) -o \\\\( /etc/krb5.keytab -nt /etc/opt/omi/creds/omi.keytab \\\\) ] && /opt/omi/bin/support/ktstrip /etc/krb5.keytab /etc/opt/omi/creds/omi.keytab >/dev/null 2>&1 || true)\",\"hostname\" : \"LinuxRedhatDesktop\",\"FluentdIngestTimestamp\" : \"2019-06-27T14:29:01Z\"},\"category\" : \"cron\",\"level\" : \"info\",\"operationName\" : \"LinuxSyslogEvent\"}",
+        "process": {
+            "args": "[ -f /etc/krb5.keytab ] && [ \\( ! -f /etc/opt/omi/creds/omi.keytab \\) -o \\( /etc/krb5.keytab -nt /etc/opt/omi/creds/omi.keytab \\) ] && /opt/omi/bin/support/ktstrip /etc/krb5.keytab /etc/opt/omi/creds/omi.keytab >/dev/null 2>&1 || true",
```
Comment (otetard) [inline Azure/azure-linux/tests/cron_command2.json:25]:
```suggestion
            "command_line": "[ -f /etc/krb5.keytab ] && [ \\( ! -f /etc/opt/omi/creds/omi.keytab \\) -o \\( /etc/krb5.keytab -nt /etc/opt/omi/creds/omi.keytab \\) ] && /opt/omi/bin/support/ktstrip /etc/krb5.keytab /etc/opt/omi/creds/omi.keytab >/dev/null 2>&1 || true",
```

**Review by otetard — CHANGES_REQUESTED** [review]
One more thing… (sorry)

File: Azure/azure-linux/tests/cron_command1.json [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+        },
+        "azure_linux": {
+            "message": "(root) CMD (/usr/lib64/sa/sa1 1 1)"
+        },
+        "log": {"hostname": "LinuxRedhatDesktop", "level": "info"},
+        "message": "{ \"time\" : \"2019-06-27T14:50:01.0000000Z\",\"resourceId\" : \"/subscriptions/128ed5ce-4f50-4b5f-a3b0-08233b5a86b6/resourceGroups/demo.sekoia.io/providers/Microsoft.Compute/virtualMachines/LinuxRedhatDesktop\",\"properties\" : {\"ident\" : \"CROND\",\"pid\" : \"21188\",\"Ignore\" : \"syslog\",\"Facility\" : \"cron\",\"Severity\" : \"info\",\"EventTime\" : \"2019-06-27T14:50:01+0000\",\"SendingHost\" : \"localhost\",\"Msg\" : \"(root) CMD (/usr/lib64/sa/sa1 1 1)\",\"hostname\" : \"LinuxRedhatDesktop\",\"FluentdIngestTimestamp\" : \"2019-06-27T14:50:01Z\"},\"category\" : \"cron\",\"level\" : \"info\",\"operationName\" : \"LinuxSyslogEvent\"}",
+        "process": {
+            "executable": "/usr/lib64/sa/sa1",
+            "command_line": "/usr/lib64/sa/sa1 1 1",
+            "pid": 21188
```
Comment (otetard) [inline Azure/azure-linux/tests/cron_command1.json:24]: This process is not `/usr/lib64/sa/sa1`’s one, its the parent one (`CROND`’s one). I suggest to move this PID to `process.parent.pid` or to remove the information.

```suggestion
            "parent": {
                "pid": 21188
             }
```

File: Azure/azure-linux/tests/cron_command2.json [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+        "process": {
+            "command_line": "[ -f /etc/krb5.keytab ] && [ \\( ! -f /etc/opt/omi/creds/omi.keytab \\) -o \\( /etc/krb5.keytab -nt /etc/opt/omi/creds/omi.keytab \\) ] && /opt/omi/bin/support/ktstrip /etc/krb5.keytab /etc/opt/omi/creds/omi.keytab >/dev/null 2>&1 || true",
+            "pid": 16373
```
Comment (otetard) [inline Azure/azure-linux/tests/cron_command2.json:26]: Same as before, either remove this information or move to `process.parent.pid`.

**Pull request merged**